sys/time.c

```diff
@@ -0,0 +1,39 @@
+#include <errno.h>
+#include <thread.h>
+#include <time.h>
+#include <timer.h>
+#include <sync.h>
+
+int do_clock_gettime(clockid_t clk, timespec_t *tp) {
+  if (tp == NULL)
+    return -EFAULT;
```

Comment:
I fixed your embarrasing failure to recognize how we return `errno` values to userspace.